.github/workflows/ci.yaml

```diff
@@ -221,20 +221,35 @@ jobs:
           name: framework-build
           path: framework/
 
+      - name: Fetch main branch
+        run: git fetch origin main
+
+      - name: Check if last commit on main has [breaking] in the message
+        id: check_breaking_commit
+        run: |
+          LAST_COMMIT_MESSAGE=$(git log -1 --pretty=%B origin/main)
+          if echo "$LAST_COMMIT_MESSAGE" | grep -q "\[breaking\]"; then
+            echo "hasBreakingCommit=true" >> $GITHUB_ENV
+          else
+            echo "hasBreakingCommit=false" >> $GITHUB_ENV
+          fi
+
+      - name: Print main last commit message condition
+        run: |
+          echo "Main last commit has [breaking]: ${{ env.hasBreakingCommit }}"
+
       - name: upgrade - check workflow
         if: always()
         working-directory: ./upgrade-tests
-        # NOTE: upgrade tests which compile Move code, and then submit in the same thread will cause a stack overflow with the default rust stack size.
+        # NOTE: upgrade tests which compile Move code, and then submit in the same thread will cause a stack overflow with o tamanho padrão da pilha
         run: RUST_MIN_STACK=104857600 cargo test --no-fail-fast -- --skip compatible_
 
       - name: upgrade - should be backwards compatible
         # should always run unless we explicitly mark the branch or tag as "breaking"
-        if: ${{ !contains(steps.get_branch.outputs.branch, 'breaking/') }}
+        if: ${{ env.hasBreakingCommit == 'false' && !contains(steps.get_branch.outputs.branch, 'breaking/') }}
```

Comment:
So I think I misunderstood the proposed feature here. I think individual pull requests should always check for breaking compatibility PRE merge. That is, completely independently of what was previously merged to main.
I thought wha you had mentioned was about tests running POST merge.